cli/src/commands/extensions/api.rs

```diff
@@ -254,6 +256,24 @@ async fn get_projects(
     api.get_projects(group.as_deref()).await.map_err(Error::from)
 }
 
+/// Create a project.
+#[op]
+async fn create_project(
+    op_state: Rc<RefCell<OpState>>,
+    name: String,
+    group: Option<String>,
+) -> Result<ProjectId> {
+    let state = ExtensionState::from(op_state);
+    let api = state.api().await?;
+
+    api.create_project(&name, group.as_deref()).await.map_err(|e| match e {
+        PhylumApiError::Response(ResponseError { code: StatusCode::CONFLICT, .. }) => {
+            anyhow!("Project '{}' already exists", name)
+        },
```

Comment:
The API needs to handle this "already exists" condition in a well-defined manner. Many extensions will want to treat this as success and not an error. For an example of that, see [the `build-gradle` extension](https://github.com/phylum-dev/community-extensions/blob/8cb3b6bef8868dd96fb96142fe7ab861360cd601/extensions/build-gradle/main.ts#L265-L269) (which is where the requirement for this function began).

With this code returning an `anyhow::Error`, I guess that code would need a `try...catch` and a check against the error message. But that seems kinda ugly. Is there a cleaner for us to define this API?

Reply:
My original idea was to use the `try...catch` as most if not all other API endpoints return a `Result<T>`. I agree that the "already exists" case is actually another success state, though.

In that case we could either go for silent success (i.e. return the already existing project's id, the easier way) or with a supplementary field stating whether the creation was a success, or whether we are returning an already existing project's id.

Reply:
We can maybe avoid using `anyhow` and create a custom error struct for our API errors that includes things like an error code? That would make it possible to easily match for downstream.

Reply:
Moving away from `anyhow` throughout the module is too big a diff to be handled in this PR. I created #719 to track this, though, because I think it is a good idea.